---
output:
  html_document: default
---
Differential Gene Expression Data of Early Stages of Intestine Regeneration in Holothuria glaberrima
===============================================================================

**Authors:** *Joshua Medina, David Quispe* | 2020

```{r, echo=FALSE, message=FALSE, cache=FALSE}
library(BiocStyle)
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
```

# Overview

The sea cucumber _Holothuria glaberrima_ has great regenerative capability after internal organs evisceration. In this case, we are analyzing data from mesentery regenerative tissue of 1 and 3 days post evisceration (1- and 3-DPE) in order to further understand the genes that are involved in the early stages of regeneration.

***

# Process

Set the directory to work in and load required libraries.

```{r}

setwd("D:/usuarios/david/Escritorio/Tesis/results/DESeq_protocol/DESeq_newprotocol/GitDESeq/")

suppressMessages(library(DESeq2))
suppressMessages(library(gplots))
suppressMessages(library(RColorBrewer))
suppressMessages(library(EnhancedVolcano))
suppressMessages(library(tidyverse))
suppressMessages(library(ggplot2))
suppressMessages(library(dplyr))
suppressMessages(library(tidyverse))
suppressMessages(library(RColorBrewer))
suppressMessages(library(calibrate))
suppressMessages(library(dichromat))
suppressMessages(library(IDPmisc))
```


**Loading the data.**

* The variable *ddata* contains a csv file indicating the files to be used and to which group they correspond (i.e experimental or control). In our case this is going to consist of three different levels normal, 1 day post-evisceration (1DPE), and 3 days post-evisceration (3DPE).

```{r}
ddata <- read.table("data/pdata.txt", header = TRUE)
```




Generating the data set with respect to the counts. Here the count files generated with salmon are going to be used for the differential gene expression quantification.

```{r}
ddsSTAR <- DESeqDataSetFromHTSeqCount(sampleTable = ddata, directory = "counts_global", design = ~ condition)

colData(ddsSTAR)$condition <- factor(colData(ddsSTAR)$condition, levels=levels(ddata$condition))

#set a factor variable with six levels (conditions of the study). 
#Totally needed another option is to use relevel.

```




Filter transcripts with low count number. This works to eliminate noise from the different samples that have heterogeneous count number from artifact transcripts or samples with very low counts that can be discriminated.

```{r}
keep <- rowSums(counts(ddsSTAR)) >= 10
ddsSTAR <- ddsSTAR[keep,]
```




Estimating the size factors using _"median ratio method"_, dispersions (gene-wise, mean-dispersion relationship, final dispersion estimates). Generating the counts table with _counts_ function.

```{r}

dds <- estimateSizeFactors(ddsSTAR)
dds <- estimateDispersions(dds)
dds <- nbinomWaldTest(dds)
counts_table <- counts(dds, normalized=TRUE)

```




Results of contrasts between conditions (1DPE vs Normal; 3DPE vs Normal). __Note: Normal tissue is referred as non-regenerated mesentery tissue of the sea cucumber.__

```{r}
res_1DPE <- results(dds, contrast = c("condition", "1DPE", "Control"))
res_1DPE <- res_1DPE[order(res_1DPE$padj),]

res_3DPE <- results(dds, contrast = c("condition", "3DPE", "Control"))
res_3DPE <- res_3DPE[order(res_3DPE$padj),]

```




# Visualization of the results.

***

Plotting dispersion estimates.

```{r, fig.align = 'center'}
DESeq2::plotDispEsts(dds,main="Dispersion Plot")
```




Comparison of Raw Data vs normalized counts.

```{r, fig.align = 'center'}

conds <- as.vector(ddata$condition)
condcols <- brewer.pal(n = length(unique(conds)), name="Dark2") 
par(mfrow=c(1,2))
barplot(colSums(counts(dds, normalized=F)), col = condcols[as.factor(conds)], las = 2, cex.names=0.6, main="Raw Counts")
barplot(colSums(counts(dds, normalized=T)), col=condcols[as.factor(conds)], las=2, cex.names = 0.6, main = "Normalized Counts")

#is normalizing really a good thing for this data?

```




PCA Plot

```{r, fig.align = 'center'}

vsd <- varianceStabilizingTransformation(dds)
DESeq2::plotPCA(vsd)+theme_bw()

```




Volcano plot of Normal vs 1DPE

```{r}
with(res_1DPE, plot(log2FoldChange, -log10(pvalue), pch=20, main="Time0 vs 1DPE", xlim=c(-30,30)))

with(subset(res_1DPE, padj<=0.05 & log2FoldChange>=2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res_1DPE, padj<=0.05 & log2FoldChange<=-2), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))

abline(h=-log10(0.05),lty=3)
abline(v=-1,lty=3)
abline(v=1,lty=3)
```




Volcano plot of Normal vs 3DPE

```{r}
with(res_3DPE, plot(log2FoldChange, -log10(pvalue), pch=20, main="Time0 vs 3DPE", xlim=c(-30,30)))

with(subset(res_3DPE, padj<=0.05 & log2FoldChange>=2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res_3DPE, padj<=0.05 & log2FoldChange<=-2), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))

abline(h=-log10(0.05),lty=3)
abline(v=-1,lty=3)
abline(v=1,lty=3)
```




# Cluster analysis

***


* The next analysis comprises count files generated with clustering of similar transcripts using Corset which hierarchically clusters the transcripts based on the distribution of shared reads and expression patterns. 

```{r}
cts<-read.table("data/counts_corset.txt", header = TRUE)
cts<- cts[,c(9,7,8,1,2,3,4,5,6)]
coldata<-read.table("data/pdata_corset.txt", row.names = 1)
```




Gene expression quantification.

```{r}
ddsSTAR_cluster<-DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ condition)
```




Setting Control as the reference level.

```{r}
colData(ddsSTAR_cluster)$condition <- factor(colData(ddsSTAR_cluster)$condition, levels=levels(coldata$condition))
```




Filtering files with low count number.

```{r}
keep_cluster <- rowSums(counts(ddsSTAR_cluster)) >= 10
ddsSTAR_cluster <- ddsSTAR_cluster[keep_cluster,]
```




Size factor estimation and dispersion. Generation of count table.

```{r}
dds_cluster <- estimateSizeFactors(ddsSTAR_cluster)
dds_cluster <- estimateDispersions(dds_cluster)
dds_cluster <- nbinomWaldTest(dds_cluster)
counts_table_cluster <- counts(dds_cluster, normalized=TRUE)
```




Contrasts between conditions (1DPE vs Normal; 3DPE vs Normal).

```{r}
res_1DPE_2 <- results(dds_cluster, contrast = c("condition", "1DPE", "Control"))
res_1DPE_2 <- res_1DPE_2[order(res_1DPE_2$padj),]

res_3DPE_2 <- results(dds_cluster, contrast = c("condition", "3DPE", "Control"))
res_3DPE_2 <- res_3DPE_2[order(res_3DPE_2$padj),]
```




Plotting dispersion estimates.

```{r}
DESeq2::plotDispEsts(dds_cluster,main="Dispersion Plot")
```




Comparison of Raw Data vs normalized counts.

```{r}
conds <- as.vector(coldata$condition)
condcols <- brewer.pal(n = length(unique(conds)), name="Dark2") 
par(mfrow=c(1,2))
barplot(colSums(counts(dds_cluster, normalized=F)), col = condcols[as.factor(conds)], las = 2, cex.names=0.6, main="Raw Counts")
barplot(colSums(counts(dds_cluster, normalized=T)), col=condcols[as.factor(conds)], las=2, cex.names = 0.6, main = "Normalized Counts")
```




PCA plot.

```{r}
vsd_cluster <- varianceStabilizingTransformation(dds_cluster)
DESeq2::plotPCA(vsd_cluster)+theme_bw()
```




Volcano plot of Normal vs 1DPE.

```{r}
with(res_1DPE_2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Time0 vs 1DPE", xlim=c(-30,30)))

with(subset(res_1DPE_2, padj<=0.05 & log2FoldChange>=2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res_1DPE_2, padj<=0.05 & log2FoldChange<=-2), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))

abline(h=-log10(0.05),lty=3)
abline(v=-1,lty=3)
abline(v=1,lty=3)
```




Volcano plot of Normal vs 3DPE.

```{r}
with(res_3DPE_2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Time0 vs 3DPE", xlim=c(-30,30)))

with(subset(res_3DPE_2, padj<=0.05 & log2FoldChange>=2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res_3DPE_2, padj<=0.05 & log2FoldChange<=-2), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))

abline(h=-log10(0.05),lty=3)
abline(v=-1,lty=3)
abline(v=1,lty=3)
```
